go/chat/mobilepush.go

```diff
@@ -75,79 +73,26 @@ func (h *MobilePush) AckNotificationSuccess(ctx context.Context, pushIDs []strin
-			msg.Valid().MessageBody.Text().Body), nil
-	default:
-		return fmt.Sprintf("%s: %s", msg.Valid().SenderUsername, msg.Valid().MessageBody.Text().Body), nil
-	}
-}
+func (h *MobilePush) GetConvInfo(ctx context.Context, uid gregor1.UID, convID chat1.ConversationID) (info chat1.ConversationInfoLocal, err error) {
+	ib, _, err := h.G().InboxSource.Read(ctx, uid, types.ConversationLocalizerBlocking,
+		types.InboxSourceDataSourceAll, nil,
+		&chat1.GetInboxLocalQuery{
+			ConvIDs: []chat1.ConversationID{convID},
+		}, nil)
 
-func (h *MobilePush) formatReactionPush(ctx context.Context, uid gregor1.UID, convID chat1.ConversationID,
-	membersType chat1.ConversationMembersType, msg chat1.MessageUnboxed) (res string, err error) {
-	reaction, err := utils.GetReaction(msg)
-	if err != nil {
-		return res, err
-	}
-	switch membersType {
-	case chat1.ConversationMembersType_TEAM:
-		// Try to get the channel name
-		ib, _, err := h.G().InboxSource.Read(ctx, uid, types.ConversationLocalizerBlocking,
-			types.InboxSourceDataSourceAll, nil,
-			&chat1.GetInboxLocalQuery{
-				ConvIDs: []chat1.ConversationID{convID},
-			}, nil)
-		if err != nil || len(ib.Convs) == 0 {
-			h.Debug(ctx, "FormatPushText: failed to unbox convo, using team only")
-			return emoji.Sprintf("(%s): %s reacted to your message with %v", msg.Valid().ClientHeader.TlfName,
-				msg.Valid().SenderUsername, reaction), nil
-		}
-		return emoji.Sprintf("(%s#%s): %s reacted to your message with %v", msg.Valid().ClientHeader.TlfName,
-			ib.Convs[0].Info.TopicName, msg.Valid().SenderUsername, reaction), nil
-	default:
-		return emoji.Sprintf("%s reacted to your message with %v", msg.Valid().SenderUsername,
-			reaction), nil
+	if err != nil || len(ib.Convs) == 0 {
+		h.Debug(ctx, "getConvInfo: failed to unbox conv: %v", convID)
+		return chat1.ConversationInfoLocal{}, fmt.Errorf("Failed to unbox %v", convID)
 	}
+
+	return ib.Convs[0].Info, nil
 }
 
-func (h *MobilePush) FormatPushText(ctx context.Context, uid gregor1.UID, convID chat1.ConversationID,
-	membersType chat1.ConversationMembersType, msg chat1.MessageUnboxed) (res string, err error) {
-	defer h.Trace(ctx, func() error { return err }, "FormatPushText: convID: %v", convID)()
-	if !msg.IsValid() {
-		h.Debug(ctx, "FormatPushText: message is not valid")
-		return res, errors.New("invalid message")
-	}
-	switch msg.GetMessageType() {
-	case chat1.MessageType_TEXT:
-		return h.formatTextPush(ctx, uid, convID, membersType, msg)
-	case chat1.MessageType_REACTION:
-		return h.formatReactionPush(ctx, uid, convID, membersType, msg)
-	default:
-		h.Debug(ctx, "FormatPushText: unknown message type: %v", msg.GetMessageType())
-		return res, errors.New("invalid message type for plaintext")
+func GetUserAvatar(username string, srvInfo *keybase1.HttpSrvInfo) string {
+	if srvInfo != nil {
+		return fmt.Sprintf("http://%v/av?typ=user&name=%v&format=square_192&token=%v", srvInfo.Address, username, srvInfo.Token)
```

Comment:
Should make this a function of the `avatars` package, probably by making that `NewSrv` thing get saved somewhere and be accessible. Not a huge deal though, so if it is annoying just skip it.